.github/workflows/coverage.yml

```
@@ -87,9 +94,46 @@ jobs:
             echo "  \"message\": \"${{ env.percentage }}%\","
             echo "  \"color\": \"${{ steps.design.outputs.colour }}\""
             echo "}"
-          } > badge.json
-      - name: Write Gist
+          } > $RUNNER_TEMP/badge.json
+      - name: Commit badge (with signature)
+        # If another workflow is running in parallel, a race condition may occur. Try to
+        # push the updated badge three times before failing.
         env:
-          gist_id: 51dd332be75961a7dc903c67718028e1
-          out_name: coreax_coverage.json
-        run: gh gist edit ${{ env.gist_id }} -a ${{ env.out_name }} badge.json
+          BADGE_PATH: coverage/coreax_coverage.json
+          GH_TOKEN: ${{ steps.generate-token.outputs.token }}
+        run: |
+          export message="chore: update coverage for $GITHUB_SHA"
+          export content=$( base64 -i $RUNNER_TEMP/badge.json )
+          # Create new file if does not exist yet (or did not exist at checkout)
+          if [ ! -f $BADGE_PATH ]; then
+            gh api --method PUT repos/:owner/coreax-metadata/contents/$BADGE_PATH \
+              -f message="$message" \
+              -f content="$content"
+            if [ $? -eq 0 ]; then
+              echo "Coverage badge created."
+              exit 0
+            fi
+          fi
+          # Update existing file, trying 3 times
+          for i in {1..3}; do
+            # Check whether file has changed
+            diff $BADGE_PATH $RUNNER_TEMP/badge.json
+            if [ $? -eq 0 ]; then
+              echo "Coverage badge unchanged."
+              exit 0
+            fi
+            # Changed: replace existing file, trying 3 times
+            export sha=$( git rev-parse main:$BADGE_PATH )
+            gh api --method PUT $BADGE_PATH \
```

Should this be
```suggestion
            gh api --method PUT repos/:owner/coreax-metadata/contents/$BADGE_PATH \
```
as above?

---

Yes